blockchain/pallets/minting/src/tests.rs

```diff
@@ -282,6 +287,65 @@ mod register_identity {
         });
     }
 
+    #[test]
+    fn first_call_to_register_for_minting_is_free() {
+        use frame_support::pallet_prelude::Pays;
+
+        new_test_ext().execute_with(|| {
+            register_id_account(42, 1);
+
+            let tree_0 = gen_tree(&["a", "b"]);
+            assert_eq!(
+                FractalMinting::register_for_minting(Origin::signed(1), Some(42), tree_0),
+                Ok(PostDispatchInfo {
+                    actual_weight: None,
+                    pays_fee: Pays::No
+                })
+            );
+        });
```

Comment:
I think it would be better to reuse the existing `register_for_minting_dataset` function. You can change that function to return the PostDispatchInfo so you can check `Pays`.

Then these lines would be simply:

```
let post = register_for_minting_dataset(1, ["a", "b"]);
assert_eq!(post.pays_fee, Pays::No);
assert_eq!(post.actual_weight, None);
```

Reply:
Agree, changed